### ISSUE TITLE
No launch paths found for resource

### ISSUE THREAD

**Issue by TiltedTower** [aws/aws-cdk#8404]
When attempting to launch a product using the CloudFormationProvisionedProduct, I keep getting an error message saying that "no launch paths" exist for the given resource.  Both the parent and the product it depends on have launch constraints on them, and when I call `aws servicecatalog list-launch-paths --product-id <product_id>` for both, they return a launch path.  Is there an explicit way to create/update a product launch path other than associating principals with portfolios?  In the end, I am trying to have a product explicitly depend on another one, and to automatically launch it if it hasn't already been deployed.  

### Reproduction Steps

Include an already existing product in a separate products CDK code like this:

```   
new SC.CfnCloudFormationProvisionedProduct(this, 'provisioned-product', {
     productId: 'prod-abcdef',
     provisioningArtifactId: 'pa-abcdef',
});
```

This was able to locate the product successfully, but it fails during product launch with the error message below.

### Error Log

`StatusReason : Unable to launch provisioned product because: No launch paths found for resource`

### Environment

  - **CLI Version      :**1.43
  - **Framework Version:**Typescript
  - **Node.js Version:**v14.4.0

--- 

This is :bug: Bug Report

**Comment by MrArnoldPalmer** [comment]
@TiltedTower so for my understanding, the failure is occurring during the cloudformation stack deploy right?

Is it possible to get the result your're looking for with regular yaml CFN templates or is this a shortcoming a the cloudformation API?

**Comment by TiltedTower** [comment]
Correct, this is only occurring during deployment.  I have been using service catalog to deploy the templates, but they're still only "regular" CFN templates that are synthesized as usual via the CDK.

**Comment by MrArnoldPalmer** [comment]
From first glance this looks like either a bug in CFN or some dependent resource is missing that causes the deploy to fail (ie something that owns the launch path). More details/investigation are required.

Since this relates to L1 constructs, which are 1:1 mappings of the cfn api, the cloudformation resource docs may have some clues. I’ll look into this when I can.

**Comment by southworthy** [comment]
I recently got this to work, although in my case I was trying to use CloudFormation to create new AWS Accounts via the "Account Factory" product. Was running into the same error message; managed to get it working by explicitly passing a launch path (`PathId` paramter of the `CloudFormationProvisionedProduct` resource).

I got the value for `PathId` using the following AWS CLI command (see `.ProductViewSummary.LaunchPaths[0]`)

```
$ aws servicecatalog describe-product --id prod-abcdef --region ap-southeast-2
{
    "ProductViewSummary": {
        "Id": "prodview-abcdef",
        "ProductId": "prod-abcdef",
        "Name": "AWS Control Tower Account Factory",
        "Owner": "AWS Control Tower",
        "ShortDescription": "AWS Control Tower Factory product. Provisions a new AWS Control Tower managed Account.",
        "Type": "CLOUD_FORMATION_TEMPLATE",
        "HasDefaultPath": false
    },
    "ProvisioningArtifacts": [
        {
            "Id": "pa-abcdef",
            "Name": "AWS Control Tower Account Factory",
            "Description": "AWS Control Tower Factory product. Provisions a new AWS Control Tower managed Account.",
            "CreatedTime": 1611622186.0,
            "Guidance": "DEFAULT"
        }
    ],
    "Budgets": [],
    "LaunchPaths": [
        {
            "Id": "lpv2-abcdef",
            "Name": "AWS Control Tower Account Factory Portfolio"
        }
    ]
}
```

In my case, there was only 1 launch path for the product, so I just used the specified ID.

I guess you could try that command and add the path ID to the config?

```
new SC.CfnCloudFormationProvisionedProduct(this, 'provisioned-product', {
     productId: 'prod-abcdef',
     provisioningArtifactId: 'pa-abcdef',
     pathId: 'lpv2-abcdef'
});
```

**Comment by YakDriver** [comment]
The error message `Unable to launch provisioned product because: No launch paths found for resource` isn't super helpful. It can mean any of the following:

1. The product doesn't exist
2. The provisioning artifact doesn't exist
3. The product exists but it's in a failed state
4. You don't have access to the product
5. You don't have access to the product's portfolio
6. The product isn't associated with a portfolio
7. The launch path does not exist

Since the error message is not helpful, it doesn't tell you which of these are to blame.

To see how unhelpful the error message is, try this for fun:

```console
% aws servicecatalog provision-product --provisioned-product-name no --product-id nope --provisioning-artifact-id nopity-nope

An error occurred (ResourceNotFoundException) when calling the ProvisionProduct operation: No launch paths found for resource: nope
```

Some pointers to getting it to work:

* Associate the product to a portfolio.
* Associate a principal that is or includes you to the portfolio.
* Make sure the product is properly created by not using `DisableTemplateValidation`. When you create the product, you'll get an error if the template has an error.
* Try describing the provisioning artifact to make sure it exists. 
* Try describing the product. If you can describe the product, it exists, and you have access. You should see a launch path as part of the product description. If you can describe the product but it doesn't have a launch path, I suspect the template is bad.

**Comment by arcrank** [comment]
Following up here, are you still encountering this issue?

**Comment by paulmarsicloud** [comment]
I recently ran into this error and even though I could run `describe-product-as-admin` CLI commands, the `provision-product` command returned an error:
```
An error occurred (ResourceNotFoundException) when calling the ProvisionProduct operation: No launch paths found for resource
```

The solution: add the IAM User/Group/Role to the Portfolio 🤦‍♂️ https://docs.aws.amazon.com/servicecatalog/latest/adminguide/catalogs_portfolios_users.html

The IAM user I was using had Administrator access and it **still** needed to be added as a user to the Service Catalog Portfolio for the product

**Comment by MrArnoldPalmer** [comment]
@paulmarsicloud that seems like the solution here. Reopen if anyone else is unable to get around this as described.
